backend/lib/models/NotifyPreference.js

```diff
@@ -0,0 +1,39 @@
+const { Schema, model } = require("mongoose");
+
+const notifyPreferenceSchema = new Schema({
+  message: {
+    instant: { default: true, type: Boolean },
+    daily: { default: false, type: Boolean },
+    weekly: { default: false, type: Boolean },
+    biweekly: { default: false, type: Boolean },
+  },
+  like: {
+    instant: { default: false, type: Boolean },
+    daily: { default: false, type: Boolean },
+    weekly: { default: true, type: Boolean },
+    biweekly: { default: false, type: Boolean },
+  },
+  comment: {
+    instant: { default: false, type: Boolean },
+    daily: { default: true, type: Boolean },
+    weekly: { default: false, type: Boolean },
+    biweekly: { default: false, type: Boolean },
+  },
+  share: {
+    instant: { default: false, type: Boolean },
+    daily: { default: true, type: Boolean },
+    weekly: { default: false, type: Boolean },
+    biweekly: { default: false, type: Boolean },
+  },
```

Comment:
I think we should add an `enabled` field, this way we can preserve the old values when the user disables email notifications completely, and can just enable them later without having to recheck the boxes to where they want them.

Reply:
that sounds like a user experience feature, not strongly relevant to our model design, I wonder if we have another way to do that without modifying our model ( ⚆ _ ⚆ )